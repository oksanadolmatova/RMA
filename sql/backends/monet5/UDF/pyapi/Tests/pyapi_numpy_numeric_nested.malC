
a:= bat.new(:dbl);
bat.append(a, 1.4:dbl);
bat.append(a, 3.4:dbl);
bat.append(a, 2.1:dbl);
bat.append(a, 8.2:dbl);
bat.append(a, 9.6:dbl);

b:= bat.new(:dbl);
bat.append(b, 13.4:dbl);
bat.append(b, 17.6:dbl);
bat.append(b, 12.8:dbl);
bat.append(b, 19.1:dbl);
bat.append(b, 19.9:dbl);

c:= bat.new(:dbl);
bat.append(c, 18.4:dbl);
bat.append(c, 18.5:dbl);
bat.append(c, 18.6:dbl);
;bat.append(c, 18.7:dbl);
bat.append(c, 18.8:dbl);

d:= bat.new(:dbl);
bat.append(d, 0.04:dbl);
bat.append(d, 0.005:dbl);
bat.append(d, 0.008:dbl);
bat.append(d, 0.04:dbl);
bat.append(d, 0.02:dbl);


# square an array
sqr:bat[:dbl] := pyapi.eval(nil:ptr, "return(arg1*arg1)", a);
io.print(sqr);

# then take the root of that array again
a:bat[:dbl] := pyapi.eval(nil:ptr, "return(numpy.sqrt(arg1))", sqr);
io.print(a);

# do some numpy operations
x:bat[:dbl] := pyapi.eval(nil:ptr, "return(numpy.prod(arg1*arg2*arg3*arg4)*numpy.sum(arg1*arg2*arg3*arg4))", a, b, c, d);
io.print(x);

fib:= bat.new(:int);
bat.append(fib, 0:int);
bat.append(fib, 1:int);
bat.append(fib, 2:int);
bat.append(fib, 3:int);
bat.append(fib, 4:int);
bat.append(fib, 5:int);
bat.append(fib, 6:int);
bat.append(fib, 7:int);
bat.append(fib, 8:int);
bat.append(fib, 9:int);
bat.append(fib, 10:int);
bat.append(fib, 11:int);
bat.append(fib, 12:int);
bat.append(fib, 13:int);
bat.append(fib, 14:int);
bat.append(fib, 15:int);
bat.append(fib, 16:int);
bat.append(fib, 17:int);
bat.append(fib, 18:int);
bat.append(fib, 19:int);
bat.append(fib, 20:int);
bat.append(fib, 21:int);
bat.append(fib, 22:int);
bat.append(fib, 23:int);
bat.append(fib, 24:int);
bat.append(fib, 25:int);
bat.append(fib, 26:int);
bat.append(fib, 27:int);
bat.append(fib, 28:int);
bat.append(fib, 29:int);

# define a function within the main function and map it to a BAT
s:bat[:dbl] := pyapi.eval(nil:ptr, "def fibonacci(nmbr):\n\tif (nmbr == 0): return 0\n\tif (nmbr == 1): return 1\n\ta = 0\n\tb = 1\n\tfor i in range(0, nmbr - 1):\n\t\tc = a + b\n\t\ta = b\n\t\tb = c\n\treturn b\nreturn(list(map(fibonacci, arg1)))", fib);
io.print(s);
