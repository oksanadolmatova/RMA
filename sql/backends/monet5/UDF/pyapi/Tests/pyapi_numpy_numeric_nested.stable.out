stdout of test 'pyapi_numpy_numeric_nested` in directory 'sql/backends/monet5/UDF/pyapi` itself:


# 00:29:49 >  
# 00:29:49 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=32717" "--set" "mapi_usock=/var/tmp/mtest-30800/.s.monetdb.32717" "--set" "monet_prompt=" "--forcemito" "--set" "mal_listing=2" "--dbpath=/usr/local/var/MonetDB/mTests_monetdb5_extras_pyapi" "--set" "mal_listing=2" "--set" "embedded_r=true" "--set" "embedded_py=true"
# 00:29:49 >  

# MonetDB 5 server v11.20.0
# This is an unreleased version
# Serving database 'mTests_monetdb5_extras_pyapi', using 8 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs and 128bit integers dynamically linked
# Found 7.684 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2015 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://mytherin-N750JV:32717/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-30800/.s.monetdb.32717
# MonetDB/SQL module loaded
# MonetDB/Python module loaded
# MonetDB/R   module loaded

Ready.

# 00:29:50 >  
# 00:29:50 >  "mclient" "-lmal" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-30800" "--port=32717"
# 00:29:50 >  

#io.print(sqr);
#--------------------------#
# h	t  # name
# void	dbl  # type
#--------------------------#
[ 0@0,	1.96	]
[ 1@0,	11.56	]
[ 2@0,	4.41	]
[ 3@0,	67.24	]
[ 4@0,	92.16	]
#io.print(a);
#--------------------------#
# h	t  # name
# void	dbl  # type
#--------------------------#
[ 0@0,	1.4	]
[ 1@0,	3.4	]
[ 2@0,	2.1	]
[ 3@0,	8.2	]
[ 4@0,	9.6	]
#io.print(x);
#--------------------------#
# h	t  # name
# void	dbl  # type
#--------------------------#
[ 0@0,	546183377.8	]
#io.print(s);
#--------------------------#
# h	t  # name
# void	dbl  # type
#--------------------------#
[ 0@0,	0	]
[ 1@0,	1	]
[ 2@0,	1	]
[ 3@0,	2	]
[ 4@0,	3	]
[ 5@0,	5	]
[ 6@0,	8	]
[ 7@0,	13	]
[ 8@0,	21	]
[ 9@0,	34	]
[ 10@0,	55	]
[ 11@0,	89	]
[ 12@0,	144	]
[ 13@0,	233	]
[ 14@0,	377	]
[ 15@0,	610	]
[ 16@0,	987	]
[ 17@0,	1597	]
[ 18@0,	2584	]
[ 19@0,	4181	]
[ 20@0,	6765	]
[ 21@0,	10946	]
[ 22@0,	17711	]
[ 23@0,	28657	]
[ 24@0,	46368	]
[ 25@0,	75025	]
[ 26@0,	121393	]
[ 27@0,	196418	]
[ 28@0,	317811	]
[ 29@0,	514229	]

# 00:29:50 >  
# 00:29:50 >  "Done."
# 00:29:50 >  

